ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -84,32 +84,43 @@ end = struct
      _the_ management interface. Hosts in a pool use the IP addresses of this interface
      to communicate with each other. *)
   let start ~__context ?addr () =
+    let port = !Constants.https_port in
     let socket, stunnel_accept =
+      let bind ~localhost addr =
+        let sockaddr = Unix.ADDR_INET (addr, Constants.http_port) in
+        let desc addr sockaddr =
+          match (localhost, Unix.domain_of_sockaddr sockaddr) with
+          | false, PF_INET ->
+              string_of_int port
+          | true, PF_INET ->
+              Printf.sprintf "127.0.0.1:%i" port
+          | false, PF_INET6 ->
+              Printf.sprintf ":::%i" port
+          | true, PF_INET6 ->
+              Printf.sprintf "::1:%i" port
+          | _, PF_UNIX ->
+              failwith
+                (Printf.sprintf
+                   "can't start xapi server on a unix socket: addr=%s, \
+                    port=%i, localhost=%b"
+                   (Unix.string_of_inet_addr addr)
+                   port localhost)
+        in
+        (Xapi_http.bind sockaddr, desc addr sockaddr)
+      in
       match addr with
       | None -> (
           info "Starting new server (listening on all IP addresses)" ;
           try
-            (* Is it IPv6 ? *)
+            (* this may succeed even if IPv6 is not used, but we must try [inet6_addr_any]
```

Comment:
This was meant to catch the case where ipv6 is disabled in the kernel. The call to bind would then fail.

Reply:
I see. My hope was that `domain_of_socket` would tell us that `Unix.inet6_addr_any` actually returned a ipv4 address, but it doesn't. I'll revert back to the old behaviour (though the observable behaviour is coincidentally equivalent)

Reply:
`Unix.inet6_addr_any` is just a constant.

Reply:
Ah this makes sense now... :facepalm: 

I tried decoupling `stunnel_accept` from the creating of the `socket`, but I don't think it makes the code any better, so I've just reverted this file